tests/ecmp/test_fgnhg.py

```diff
@@ -0,0 +1,272 @@
+    duthost.shell("sonic-cfggen -j /tmp/neigh.json --write-to-db")
+
+
+def create_fg_ptf_config(ptfhost, ip_to_port, port_list, bank_0_port, bank_1_port, router_mac, net_ports, prefix):
+    fg_ecmp = {
+            "ip_to_port": ip_to_port,
+            "port_list": port_list,
+            "bank_0_port": bank_0_port,
+            "bank_1_port": bank_1_port,
+            "dut_mac": router_mac,
+            "dst_ip": prefix.split('/')[0],
+            "net_ports": net_ports
+    }
+    ptfhost.copy(content=json.dumps(fg_ecmp, indent=2), dest=FG_ECMP_CFG)
+
+
+def setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts, router_mac, net_ports, vlan_ip, prefix):
+    port_list, ip_to_port, bank_0_port, bank_1_port = configure_interfaces(cfg_facts, duthost, ptfhost, ptfadapter, vlan_ip)
+    generate_fgnhg_config(duthost, ip_to_port, bank_0_port, bank_1_port, prefix)
+    time.sleep(60)
+    setup_neighbors(duthost, ptfhost, ip_to_port)
+    create_fg_ptf_config(ptfhost, ip_to_port, port_list, bank_0_port, bank_1_port, router_mac, net_ports, prefix)
+    ptfhost.copy(src="ptftests/fg_ecmp_test.py", dest="/root/ptftests")
+    return port_list, ip_to_port, bank_0_port, bank_1_port
+
+
+def fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, prefix):
+
+    if isinstance(ipaddress.ip_network(prefix), ipaddress.IPv4Network):
+        ipcmd = "ip route"
+    else:
+        ipcmd = "ipv6 route"
+
+    for nexthop in ip_to_port:
+        duthost.shell("vtysh -c 'configure terminal' -c '{} {} {}'".format(ipcmd, prefix, nexthop))
+    time.sleep(1)
+
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.log".format(datetime.now().strftime('%Y-%m-%d-%H:%M:%S'))
+
+    ptf_runner(ptfhost,
+	   "ptftests",
+	   "fg_ecmp_test.FgEcmpTest",
+	    platform_dir="ptftests",
+	    params={"test_case": 'create_flows',
+		    "config_file": FG_ECMP_CFG},
+	    qlen=1000,
+	    log_file=log_file)
+
+    ptf_runner(ptfhost,
+	   "ptftests",
+	   "fg_ecmp_test.FgEcmpTest",
+	    platform_dir="ptftests",
+	    params={"test_case": 'initial_hash_check',
+		    "config_file": FG_ECMP_CFG},
+	    qlen=1000,
+	    log_file=log_file) 
+
+    withdraw_nh_port = bank_0_port[1]
+    for nexthop, port in ip_to_port.items():
+        if port == withdraw_nh_port:
+            duthost.shell("vtysh -c 'configure terminal' -c 'no {} {} {}'".format(ipcmd, prefix, nexthop))
+
+    time.sleep(1)
+
+    ptf_runner(ptfhost,
+	   "ptftests",
+	   "fg_ecmp_test.FgEcmpTest",
+	    platform_dir="ptftests",
+	    params={"test_case": 'withdraw_nh',
+		    "config_file": FG_ECMP_CFG,
+                    "withdraw_nh_port": withdraw_nh_port},
+	    qlen=1000,
+	    log_file=log_file)
+
+    for nexthop, port in ip_to_port.items():
+        if port == withdraw_nh_port:
+            duthost.shell("vtysh -c 'configure terminal' -c '{} {} {}'".format(ipcmd, prefix, nexthop))
+
+    time.sleep(1)
+
+    ptf_runner(ptfhost,
```

Comment:
please use ptf_adapter to send the packets.

Reply:
I had tried to use ptf_adapter, but ran into scale issues since this test sends 1000 packets. ptf_adapter appears to make a ssh call for every packet send with the ptfadapter, and with 1k packets it would just get stuck.